src/engine/pieces/piece.ts

```diff
@@ -13,8 +17,35 @@ export default class Piece {
         throw new Error('This method must be implemented, and return a list of available moves');
     }
 
+    public addAvailableMoveAndContinue(availableMoves: Array<Square>, newRow : number, newCol : number, currentPosition : Square, board : Board){
+        let newPos = new Square(newRow, newCol);
+        let newPosInBound = newPos.inBoundsCheck();
```

Comment:
these should both be `const `instead of `let`, this also applies in other places if you could have a check, whenever something does not change we want to be using `const`

Reply:
Fixed